components/Common/Badge/index.tsx

```diff
@@ -0,0 +1,55 @@
+import ArrowRightIcon from '@heroicons/react/24/solid/ArrowRightIcon';
+import Link from 'next/link';
+import type { ComponentProps, FC, PropsWithChildren } from 'react';
+import styles from './index.module.scss';
+import { component } from '@/types/component';
+import classNames from 'classnames';
+
+type BadgeProps = {
+  kind?: component;
+  badgeText?: string;
+} & ComponentProps<typeof Link>;
+
+const Badge: FC<PropsWithChildren<BadgeProps>> = ({
+  kind,
+  badgeText,
+  children,
+  ...args
+}) => {
+  if (typeof kind === 'undefined') kind = component.Default;
```

Comment:
That's far from optimal. If you want to handle the "undefined" behaviour, you should set within the prop `kind = component.Default`